packages/backend/server/src/core/doc/manager.ts

```diff
@@ -303,9 +317,56 @@ export class DocManager implements OnModuleInit, OnModuleDestroy {
             reject(new Error('Failed to push update'));
           },
         });
-    }).then(() => {
-      return this.updateCachedUpdatesCount(workspaceId, guid, updates.length);
     });
+    await this.updateCachedUpdatesCount(workspaceId, guid, updates.length);
+
+    return timestamp;
+  }
+
+  /**
+   * Get latest timestamp of all docs in the workspace.
+   */
+  @CallTimer('doc', 'get_stats')
+  async getStats(workspaceId: string, after: number | undefined = 0) {
+    const snapshots = await this.db.snapshot.findMany({
+      where: {
+        workspaceId,
+        updatedAt: {
+          gt: new Date(after),
+        },
+      },
+      select: {
+        id: true,
+        updatedAt: true,
+      },
+    });
+
+    const updates = await this.db.update.groupBy({
+      where: {
+        workspaceId,
+        createdAt: {
+          gt: new Date(after),
+        },
+      },
+      by: ['workspaceId', 'id'],
+      _max: {
+        createdAt: true,
+      },
+    });
```

Comment:
can improve by `create index idx_update_created_at on update (workspaceId, createAt desc)`

Reply:
if so, it should be `Index(workspaceId, guid, createdAt)`

Reply:
agree, add guid can speed up query for specific subdocs, but i still recommend add `desc` sorting for craeteAt -- there should be no scenario where we need to find the oldest snapshot

Reply:
that make sense, but I wondering whether it worth to add such an index, because the stats query won't be too frequent.

Reply:
and what we hope is that updates will be merged as soon as possible so there won't be too many rows in updates table, which decrease the necessity again